test/regpal/pal.cpp

```diff
@@ -0,0 +1,186 @@
+#define DNCP_DEFINE_GUID
+#include "pal.hpp"
+
+#include <nethost.h>
+#include <hostfxr.h>
+
+#ifndef BUILD_WINDOWS
+#include <dlfcn.h>
+#endif
+
+#include <iostream>
+#include <fstream>
+#include <filesystem>
+#include <string_view>
+
+using std::filesystem::path;
+
+#ifdef BUILD_WINDOWS
+#define X(str) std::wstring_view{L##str}
+#else
+#define X(str) std::string_view{str}
+#endif
```

Comment:
We should be consistent here. Perhaps use `W()`?

Reply:
The `W` macro is already defined by the Windows SDK. I'll pick another name that makes it clear that this is a string view